### PR TITLE
[OB-2804] fix: Handle non-JSON error responses

### PR DESCRIPTION
**Pull request by MAG-pw**
This adds a function to HttpPayload to check if the payload has a JSON mime-type. This is then used when printing error messages to check if an error response is JSON formatted before trying to parse it as JSON. If it is not JSON then it is simply printed as is.

connected-spaces-platform Pull Request

**For the reviewer**

* [ ] If required, are the changes covered by appropriate tests?
* [ ] Are any public-facing API changes well documented?
* [ ] Is the code easily readable and extensible and/or follow existing conventions?

### CODE REVIEW

File: Library/src/Web/HttpPayload.cpp [hunk 1]
```diff
@@ -207,4 +207,18 @@ void HttpPayload::SetBoundary(const csp::common::String& InBoundary)
 	Boundary = InBoundary;
 }
 
+const bool HttpPayload::IsJsonPayload() const
+{
+	const auto ContentTypeHeader = Headers.find("content-type");
```
Comment (MAG-mv) [inline Library/src/Web/HttpPayload.cpp:212]: Should we add a reference here to prevent a copy?

File: Library/src/Web/HttpPayload.cpp [hunk 2]
```diff
@@ -207,4 +207,18 @@ void HttpPayload::SetBoundary(const csp::common::String& InBoundary)
...
+	if (ContentTypeHeader == Headers.end())
+	{
+		return false;
+	}
+	else
+	{
+		csp::common::String ContentType = ContentTypeHeader->second.c_str();
+		return ContentType.Split(';')[0] == "application/json";
```
Comment (MAG-mv) [inline Library/src/Web/HttpPayload.cpp:220]: Do we need an aditional check here to ensure the value is valid before splitting / accessing the value?

File: Library/src/Web/HttpPayload.h [hunk 3]
```diff
@@ -71,6 +71,8 @@ class HttpPayload
 
 	void SetBoundary(const csp::common::String& InBoundary);
 
+	const bool IsJsonPayload() const;
```
Comment (MAG-mv) [inline Library/src/Web/HttpPayload.h:74]: Returning a const here inst necessary as the return value isn't a pointer / ref

**Review by MAG-mv — CHANGES_REQUESTED** [review]

**Review by MAG-SamBirley — APPROVED** [review]

**Comment by MAG-pw** [comment]
Converted back to draft as I've discovered an issue with this.

File: Library/src/Web/HttpPayload.cpp (hunk 2 above)
Comment (MAG-pw) [reply Library/src/Web/HttpPayload.cpp]: Changed how this is done so resolving.

File: Library/src/Web/WebClient.cpp [hunk 4]
```diff
@@ -380,38 +381,81 @@ void WebClient::DestroyRequest(HttpRequest* Request)
...
+		// the full response will be printed later.
 		if (ResponseJson.HasMember("errors"))
 		{
-			const auto ResponseArray = ResponseJson["errors"].GetObject().FindMember("")->value.GetArray();
+			if (ResponseJson["errors"].IsArray())
+			{
+				const auto& ResponseArray = ResponseJson["errors"].GetArray();
 
-			for (uint32_t i = 0; i < ResponseArray.Size(); i++)
+				for (uint32_t i = 0; i < ResponseArray.Size(); i++)
```
Comment (MAG-mv) [inline Library/src/Web/WebClient.cpp:416]: This doesn't look like a safe call, we have declared i twice

**Review by MAG-mv — CHANGES_REQUESTED** [review]

**Review by MAG-mv — APPROVED** [review]

**Pull request merged**